ci: add spell checker and fix existing typo

hezhizhen opened the pull request.

mehalter commented on .vscode/settings.json:
```diff
@@ -0,0 +1,4 @@
+{
+  "cSpell.words": ["astronvim"],
+  "Lua.diagnostics.globals": ["vim"]
+}
```
This file should not be committed

utkarshgupta137 replied:
Awkward... :D

hezhizhen replied:
Oh, sorry...

mehalter commented on typos.toml:
```diff
@@ -0,0 +1,6 @@
+# See https://github.com/crate-ci/typos/blob/master/docs/reference.md to configure typos
+[default.extend-words]
+enew = "enew"
+fo = "fo"
```
Could you explain these two word additions?

hezhizhen replied:
As you can see above, they are in the section of `[default.extend-words]`, which means these words are supposed to be correct (they may be not valid English words, but they are valid keywords in a certain programming language, or a commit hash, etc). You can see https://github.com/crate-ci/typos/blob/master/docs/reference.md for more infomation.

mehalter replied:
Yeah, so can you explain why these words shouldn't be considered wrong? Maybe my original question wasn't clear 😅

mehalter replied:
Is this checking the words in the code as well or just the comments?

hezhizhen replied:
Checking all

mehalter replied:
Hm I'll have to think about this and get back to you 🤔 also what about other things we have that aren't English words? Like `ffi`? Why is that not triggering it

hezhizhen replied:
> Yeah, so can you explain why these words shouldn't be considered wrong? Maybe my original question wasn't clear 😅

Oh, I see. `enew` is a vim command and `fo` is a keyboard shortcut, so I think they should be thought as correct.

mehalter replied:
There are a lot more non English words in the code than just those two. I will review this sometime next week and get back to you. Thanks for having this idea :) ideally we would be able to just spell check the comments

hezhizhen replied:
> > Yeah, so can you explain why these words shouldn't be considered wrong? Maybe my original question wasn't clear 😅
> 
> Oh, I see. `enew` is a vim command and `fo` is a keyboard shortcut, so I think they should be thought as correct.

See https://github.com/crate-ci/typos/#false-positives

mehalter replied:
Maintaining these "corrections" will get unruly and unmanageable in the future.

hezhizhen replied:
> Hm I'll have to think about this and get back to you 🤔 also what about other things we have that aren't English words? Like `ffi`? Why is that not triggering it

I haven't read its design docs and I just use it to find out potential misspelling. As far as I see, it's not something like dictionary and list all non-English-valid words.

hezhizhen replied:
> Maintaining these "corrections" will get unruly and unmanageable in the future.

We can skip some certain files in `[files]` section like `mappings.lua` to solve it.

mehalter replied:
There are many many non English words throughout all of the files. I'll get back to you in - a few days.

hezhizhen replied:
> There are many many non English words throughout all of the files. I'll get back to you in - a few days.

And we can see how https://github.com/crate-ci/typos/ works. Maybe it will answer your questions.

mehalter reviewed (approved):
I took a look into the architecture of `typos` and it looks good. It does appear that it doesn't actually check against a known dictionary so we shouldn't have to worry about adding every single missing word which is good! Thanks for the recommendation :D

hezhizhen commented:
@mehalter What have you changed in this PR?

mehalter commented:
The main thing was just rebasing after other changes got made to `nightly`. I also changed `typos.yml` to `.typos.yml` and removed the explicit definition of the configuration file from the GitHub action workflow since the configuration file can be implicit

hezhizhen commented:
> The main thing was just rebasing after other changes got made to `nightly`. I also changed `typos.yml` to `.typos.yml` and removed the explicit definition of the configuration file from the GitHub action workflow since the configuration file can be implicit

What's the benefit of renaming `typos.toml` as `.typos.toml`?

mehalter commented:
> What's the benefit of renaming `typos.toml` as `.typos.toml`?

Clean up the folder a bit by utilizing hidden files. It makes it clearer that dotfiles are there to configure tools that we use in AstroNvim's management and do not actually affect AstroNvim itself. It's more just standard practice.

hezhizhen commented:
> > What's the benefit of renaming `typos.toml` as `.typos.toml`?
> 
> Clean up the folder a bit by utilizing hidden files. It makes it clearer that dotfiles are there to configure tools that we use in AstroNvim's management and do not actually affect AstroNvim itself. It's more just standard practice.

Thanks for your explanation.
